systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -557,12 +564,28 @@ protected void convertCRDs(JsonObject conversionTool, String namespace) throws I
 
         Exec.exec("chmod", "+x", convertorPath);
 
-        LOGGER.info("Converting CRs");
+        LOGGER.info("Converting CRs ...");
         // run conversion of crs
-        Exec.exec(convertorPath, "cr", "-n=" + namespace);
+        // CRs conversion may fail, because for 0.23 for example it's already done
+        // CRS conversion needs old versions of Strimzi CRDs, which are not available after 0.22
+        if (cmdKubeClient().exec(true, false, "get", "crd", "kafkas.kafka.strimzi.io", "-o", "jsonpath={.status.storedVersions}").out().trim().contains(Constants.V1ALPHA1)) {
+            ExecResult execResult = Exec.exec(convertorPath, "cr", "-n=" + namespace);
+            LOGGER.debug("CRs conversion STDOUT:");
+            LOGGER.debug(execResult.out());
+            LOGGER.debug("CRs conversion STDERR:");
+            LOGGER.debug(execResult.err());
+            LOGGER.info("CRs conversion done!");
+        } else {
+            LOGGER.info("CRs already have v1beta2 versions");
+        }
         // run crd-upgrade
         LOGGER.info("Converting CRDs");
-        Exec.exec(convertorPath, "crd");
+        ExecResult execResult = Exec.exec(convertorPath, "crd");
+        LOGGER.debug("CRDs conversion STDOUT:");
+        LOGGER.debug(execResult.out());
+        LOGGER.debug("CRDs conversion STDERR:");
+        LOGGER.debug(execResult.err());
+        LOGGER.info("CRDs conversion done!");
```

Comment:
I don't think the logic of this works ... IMHO you should:
* Include this whole thing into the block
* You should check if it has something else than `v1beta2` => I doubt the Kafka CRD for example has ever `v1alpha1` there